pyhms/demes/initialize.py

```diff
@@ -3,61 +3,90 @@
     CMALevelConfig,
     DELevelConfig,
     EALevelConfig,
-    LHSLevelConfig,
     LocalOptimizationConfig,
-    SobolLevelConfig,
+    RandomLEvelConfig,
 )
+from pyhms.utils.parameter_calculation import get_default_mutation_std
 from structlog.typing import FilteringBoundLogger
 
 from ..core.individual import Individual
+from ..core.initializers import (
+    GaussianInitializer,
+    GaussianInitializerWithSeedInject,
+    InjectionInitializer,
+    LHSGlobalInitializer,
+    PopInitializer,
+    SobolGlobalInitializer,
+    UniformGlobalInitializer,
+)
 from .abstract_deme import AbstractDeme
 from .cma_deme import CMADeme
 from .de_deme import DEDeme
 from .ea_deme import EADeme
-from .lhs_deme import LHSDeme
 from .local_deme import LocalDeme
-from .sobol_deme import SobolDeme
+from .random_deme import RandomDeme
 
 
 def init_root(config: BaseLevelConfig, logger: FilteringBoundLogger) -> AbstractDeme:
-    return init_from_config(config, "root", 0, 0, None, logger)
+    return init_from_config(config, "root", 0, 0, None, None, logger)
 
 
 def init_from_config(
     config: BaseLevelConfig,
     new_id: str,
     target_level: int,
     metaepoch_count: int,
-    sprout_seed: Individual,
+    sprout_seed: Individual | None,
+    injected_population: list[Individual] | None,
     logger: FilteringBoundLogger,
     random_seed: int = None,
     parent_deme: AbstractDeme | None = None,
 ) -> AbstractDeme:
+    child_initializer: PopInitializer
+    if config.pop_initializer_class == UniformGlobalInitializer:
+        child_initializer = UniformGlobalInitializer(config.bounds)
+    elif config.pop_initializer_class == GaussianInitializer:
+        if hasattr(config, "sample_std_dev"):
+            sample_std_dev = config.sample_std_dev
+        else:
+            sample_std_dev = get_default_mutation_std(config.bounds, target_level)
+        child_initializer = GaussianInitializer(seed=sprout_seed.genome, std_dev=sample_std_dev, bounds=config.bounds)
+    elif config.pop_initializer_class == GaussianInitializerWithSeedInject:
+        if hasattr(config, "sample_std_dev"):
+            sample_std_dev = config.sample_std_dev
+        else:
+            sample_std_dev = get_default_mutation_std(config.bounds, target_level)
+        child_initializer = GaussianInitializerWithSeedInject(
+            seed=sprout_seed, std_dev=sample_std_dev, bounds=config.bounds
+        )
+    elif config.pop_initializer_class == LHSGlobalInitializer:
+        child_initializer = LHSGlobalInitializer(config.bounds, random_seed)
+    elif config.pop_initializer_class == SobolGlobalInitializer:
+        child_initializer = SobolGlobalInitializer(config.bounds, random_seed)
+    elif config.pop_initializer_class == InjectionInitializer:
+        if injected_population is None and sprout_seed is not None:
+            injected_population = [sprout_seed]
+        child_initializer = InjectionInitializer(injected_population, config.bounds)
```

Comment:
This should be moved to another function?

Reply:
Maybe it would be better to use match case: https://peps.python.org/pep-0622/ instead of if/elif